tools/build_executable.py

```diff
@@ -0,0 +1,108 @@
+    unreal_project_dir = os.path.realpath(os.path.join(repo_dir, "cpp", "unreal_projects", "SpearSim"))
+    unreal_plugins_dir = os.path.realpath(os.path.join(repo_dir, "cpp", "unreal_plugins"))
+
+    # remove any contents already present for a clean build
+    if os.path.exists(repo_dir):
+        shutil.rmtree(repo_dir)
+        os.makedirs(repo_dir)
+
+    if sys.platform == "win32":
+        run_uat_script = os.path.join(args.unreal_engine_dir, "Engine", "Build", "BatchFiles", "RunUAT.bat")
+        target_platform = "Win64"
+        archive_dir = os.path.join(args.output_dir, f"SpearSim-{target_platform}-{args.config_mode}")
+        cmd_prefix = f"conda activate {args.conda_env}&"
+    elif sys.platform == "darwin":
+        run_uat_script = os.path.join(args.unreal_engine_dir, "Engine", "Build", "BatchFiles", "RunUAT.sh")
+        target_platform = "Mac"
+        archive_dir = os.path.join(args.output_dir, f"SpearSim-{target_platform}-{args.config_mode}-Unsigned")
+        conda_script = args.conda_script
+        if not conda_script:
+            conda_script = "~/anaconda3/etc/profile.d/conda.sh"
+        cmd_prefix = f". {conda_script}; conda activate {args.conda_env};"
+    elif sys.platform == "linux":
+        run_uat_script = os.path.join(args.unreal_engine_dir, "Engine", "Build", "BatchFiles", "RunUAT.sh")
+        target_platform = "Linux"
+        archive_dir = os.path.join(args.output_dir, f"SpearSim-{target_platform}-{args.config_mode}")
+        conda_script = args.conda_script
+        if not conda_script:
+            conda_script = "~/anaconda3/etc/profile.d/conda.sh"
+        cmd_prefix = f". {conda_script}; conda activate {args.conda_env};"
+    else:
+        assert False
+
+    # clone repo along with submodules
+    cmd = ["git", "clone", "--recurse-submodules", "https://github.com/isl-org/spear", repo_dir]
+    print(f"[SPEAR | build_executable.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd, check=True)
+    assert cmd_result.returncode == 0
+
+    if args.commit_id:
+        cmd = ["git", "reset", "--hard", args.commit_id]
+        print(f"[SPEAR | build_executable.py] Executing: {' '.join(cmd)}")
+        cmd_result = subprocess.run(cmd, check=True)
+        assert cmd_result.returncode == 0
+
+    # build third-party libs
+    cmd = ["python", "build_third_party_libs.py", "--num_parallel_jobs", f"{args.num_parallel_jobs}"]
+    print(f"[SPEAR | build_executable.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd, check=True)
+    assert cmd_result.returncode == 0
+
+    # create symbolic links
+    cmd = cmd_prefix + f"python create_symbolic_links.py --unreal_project_dir {unreal_project_dir} --unreal_plugins_dir {unreal_plugins_dir}"
+    print(f"[SPEAR | build_executable.py] Executing: {cmd}")
+    cmd_result = subprocess.run(cmd, shell=True, check=True)
+    assert cmd_result.returncode == 0
```

Comment:
if check=True, do we need assert cmd_result.returncode == 0?

Reply:
We don't need the assert statement. I removed it.